File: 04-DFS&BFS/1260_DFS와BFS.py
n,m,v = map(int,input().split())
visited = [0] * (n+1) 
maps = list([0]*(n+1) for i in range(n+1))

for i in range(m):
    a,b = map(int,input().split())
    maps[a][b] = maps[b][a] = 1             # 간선을 연결한 맵 만들기
    
def dfs(v):
    visited[v] = 1 
    print(v, end=' ')                        # 띄어쓰기가 있으므로 end = '' 사용
    for i in range(1, n+1):
        if visited[i] == 0 and maps[v][i] == 1: # 방문하지 않고 연결된 노드 일 경우
            dfs(i)                          # 재귀

def bfs(v):
    queue = [v] 
    visited[v] = 0 
    while queue:
        v = queue.pop(0) 
        print(v,end=' ') 
        for i in range(1 , n+1):
            if(visited[i] == 1 and maps[v][i] == 1):# 방문하지 않고 연결된 노드 일 경우
                queue.append(i)
                visited[i] = 0

dfs(v)
print()                                         # 사이에 엔터...역할..
bfs(v)

File: 04-DFS&BFS/1260_DFS와BFS.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
#include <queue>
#include <memory.h>
#define  ll long long
using namespace std;
bool visited[10001];
vector<int> adj[10001];
queue<int> q;
int N, M, V;
void bfs(int now)
{
	visited[now] = 1;
	q.push(now);

	while (!q.empty()) {
		int n = q.front();
		q.pop();
		cout << n << " ";

		for (int i = 0; i < adj[n].size(); i++) {
			int next = adj[n][i];
			if (!visited[next]) {
				visited[next] = 1;
				q.push(next);
			}
		}
	}
}

void dfs(int now)
{
	visited[now] = 1;
	cout << now << " ";

	for (int i = 0; i < adj[now].size(); i++)
	{

		if (!visited[adj[now][i]])
		{
			dfs(adj[now][i]);
		}
	}
}

int main()
{
	int a, b;
	cin >> N >> M >> V;

	for (int i = 0; i < M; i++) {
		cin >> a >> b;
		adj[a].push_back(b);
		adj[b].push_back(a);
	}

	for (int index = 0; index <= N; index++)
	{
		if (adj[index].size() > 1) sort(adj[index].begin(), adj[index].end());
	}

	dfs(V);
	memset(visited, 0, 10001 * sizeof(bool));
	cout << endl;
	bfs(V);



	return 0;
}